tests/test_routing_pick_runtime.py

```
@@ -0,0 +1,251 @@
+    assert len(result) == 2
+
+
+def test_pick_runtime_returns_adapter_when_available():
+    """
+    Verifies that _pick_runtime selects and returns the registered adapter when its health indicates it is available.
+    
+    The test seeds the engine's health cache with the stub adapter marked available and asserts the returned adapter is the same stub instance.
+    """
+    stub = _StubAdapter()
+    engine = _make_engine([stub], {stub.RUNTIME_ID: True})
+    adapter, candidates_info = engine._pick_runtime("code_generation", None)
```

Ruff (0.15.12)

[warning] Unpacked variable `candidates_info` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)